test/components/portal.spec.tsx

```diff
@@ -76,36 +79,37 @@ describe('<Portal />', function() {
             container
         );
 
-        await waitFor(() => expect(bodySelect('UPDATED_SPAN')).to.be.present());
+        await waitFor(() => expect(portal.content[0]).to.be.present());
     });
 
     it('renders the portal in the bottom of the DOM', async function() {
-        const {container} = clientRenderer.render(
+        const {container, result} = clientRenderer.render(
             <Portal>
                 <span data-automation-id="SPAN">Portal Body</span>
             </Portal>
         );
 
-        await waitFor(() => {
-            const portal = bodySelect<HTMLElement>(portalId)!;
-            const children = bodySelect<HTMLElement>('SPAN')!;
+        const portal = new PortalTestDriver(result as Portal);
 
+        await waitFor(() => {
             /* tslint:disable:no-bitwise */
-            expect(portal.compareDocumentPosition(children) & Node.DOCUMENT_POSITION_CONTAINED_BY,
+            expect(portal.root.compareDocumentPosition(portal.content[0]) & Node.DOCUMENT_POSITION_CONTAINED_BY,
                 'children contained in portal').to.equal(Node.DOCUMENT_POSITION_CONTAINED_BY);
-            expect(container.compareDocumentPosition(portal) & Node.DOCUMENT_POSITION_FOLLOWING,
+            expect(container.compareDocumentPosition(portal.root) & Node.DOCUMENT_POSITION_FOLLOWING,
                 'portal is following the app container').to.equal(Node.DOCUMENT_POSITION_FOLLOWING);
             /* tslint:enable:no-bitwise */
         });
     });
 
     it('renders with a className passed as a prop', async function() {
-        clientRenderer.render(
+        const {result} = clientRenderer.render(
             <Portal className="test-class">
                 <span data-automation-id="SPAN">Portal Body</span>
             </Portal>
         );
 
-        await waitFor(() => expect(bodySelect(portalId)!.className).to.contain('test-class'));
+        const portal = new PortalTestDriver(result as Portal);
+
+        await waitFor(() => expect((portal.root as Element).className).to.contain('test-class'));
```

Comment:
If we don't expect the portal to have any other class names, we should check `.equals('test-class')`. I'm not even sure we need this test, since the root element contract should catch this.

Reply:
contract does not apply for portal because its a unique case